sdk/samples/iot/aziot_realtek_amebaD/aziot_realtek_amebaD.ino

```diff
@@ -0,0 +1,385 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <string.h>
+#include <stdbool.h>
+#include <NTPClient.h>
+#include <sntp/sntp.h>
+
+#include <cstdlib>
+
+#include <WiFi.h>
+#include <PubSubClient.h>
+
+#include <mbedtls/base64.h>
+#include <mbedtls/sha256.h>
+
+#include <az_result.h>
+#include <az_span.h>
+#include <az_iot_hub_client.h>
+
+#include "iot_configs.h"
+#include "ca.h"
+
+// Status LED: will remain high on error and pulled high for a short time for each successful send.
+#define LED_PIN 2
+#define sizeofarray(a) (sizeof(a) / sizeof(a[0]))
+#define MQTT_PACKET_SIZE 1024
+
+static const char* ssid = IOT_CONFIG_WIFI_SSID;
+static const char* password = IOT_CONFIG_WIFI_PASSWORD;
+static const char* host = IOT_CONFIG_IOTHUB_FQDN;
+static const char* device_id = IOT_CONFIG_DEVICE_ID;
+static const az_span device_key = AZ_SPAN_FROM_STR(IOT_CONFIG_DEVICE_KEY);
+static const int port = 8883;
+
+static WiFiUDP ntp_udp_client;
+static WiFiSSLClient wifi_client;
+static PubSubClient mqtt_client(wifi_client);
+static az_iot_hub_client hub_client;
+static char sas_token[200];
+static size_t sas_token_length;
+
+static uint8_t signature[512];
+static unsigned long next_telemetry_send_time_ms = 0;
+static char telemetry_topic[128];
+static uint8_t telemetry_payload[100];
+static uint32_t telemetry_send_count = 0;
+static unsigned char* ca_pem_nullterm;
+
+extern "C"{
+  extern int mbedtls_base64_decode( unsigned char *dst, size_t dlen, size_t *olen, const unsigned char *src, size_t slen);
+  extern int rom_hmac_sha256(const u8 *key, size_t key_len, const u8 *data, size_t data_len, u8 *mac);
+  extern int mbedtls_base64_encode( unsigned char *dst, size_t dlen, size_t *olen, const unsigned char *src, size_t slen);
+}
+static void createNullTerminatedRootCert()
+{
+  ca_pem_nullterm = (unsigned char*)malloc(ca_pem_len + 1);
+
+  if (ca_pem_nullterm == NULL)
+  {
+    Serial.println("Failed allocating memory for null-terminated root ca");
+  }
+  
+  memcpy(ca_pem_nullterm, ca_pem, ca_pem_len);
+  ca_pem_nullterm[ca_pem_len] = '\0';
+}
+
+
+static void connectToWiFi()
+{
+  Serial.begin(115200);
+  Serial.println();
+
+  Serial.print("Connecting to WIFI SSID ");
+  Serial.println(ssid);
+
+  WiFi.begin((char*)ssid, (char*)password);
+  while (WiFi.status() != WL_CONNECTED)
+  {
+    delay(500);
+    Serial.print(".");
+  }
+
+  Serial.println("");
+  Serial.print("WiFi connected, IP address: ");
+  Serial.println(WiFi.localIP());
+}
+
+void receivedCallback(char* topic, byte* payload, unsigned int length)
+{
+  Serial.print("Received [");
+  Serial.print(topic);
+  Serial.print("]: ");
+  for (unsigned int i = 0; i < length; i++)
+  {
+    Serial.print((char)payload[i]);
+  }
+}
+
+static void initializeClients()
+{
+  Serial.println("Initializing MQTT client");
+  
+  wifi_client.setRootCA(ca_pem_nullterm);
```

Comment:
Why did you choose to use wifi_client.setRootCA instead of wifi_client.setCACert here?

Reply:
This was the recommended code from the samples provided by Realtek.